shared/src/main/scala/mlscript/Typer.scala

```diff
@@ -263,7 +267,7 @@ class Typer(var dbg: Boolean, var verbose: Bool, var explainErrors: Bool) extend
       case p: ProxyType => fieldsOf(p.underlying, paramTags)
       case Without(base, ns) => fieldsOf(base, paramTags).filter(ns contains _._1)
       case TypeBounds(lb, ub) => fieldsOf(ub, paramTags)
-      case _: ObjectTag | _: FunctionType | _: TupleType | _: TypeVariable
+      case _: ObjectTag | _: FunctionType | _: ArrayBase | _: TypeVariable
```

Comment:
Add in the doc of the method: `Note: this does not retrieve the positional fields implicitly defined by tuples`

Reply:
I think I would just resolve this conversion :P

Reply:
Did you not understand that change I was proposing? You should try to apply my suggestions and if you're not sure how ask for clarification.

Reply:
Is it adding this comment to the code? @_@ (not sure

Reply:
I said "in the doc of the method", which means here, after this line: https://github.com/hkust-taco/mlscript/blob/11a2ba2505f05aabfeefff558cd01557c0dc8810/shared/src/main/scala/mlscript/Typer.scala#L258

Reply:
I see